### ISSUE TITLE
Soil download error: 'en' is an unexpected token

### ISSUE THREAD

**Issue by BrianCollinss** [APSIMInitiative/ApsimX#9282]
### What happened?

```
System.Xml.XmlException: 'en' is an unexpected token. The expected token is '"' or '''. Line 1, position 27.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.Throw(String res, String[] args)
   at System.Xml.XmlTextReaderImpl.ThrowUnexpectedToken(String expectedToken1, String expectedToken2)
   at System.Xml.XmlTextReaderImpl.ParseAttributes()
   at System.Xml.XmlTextReaderImpl.ParseElement()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XmlLoader.LoadDocSequence(XmlDocument parentDoc)
   at System.Xml.XmlLoader.Load(XmlDocument doc, XmlReader reader, Boolean preserveWhitespace)
   at System.Xml.XmlDocument.Load(XmlReader reader)
   at System.Xml.XmlDocument.Load(Stream inStream)
   at UserInterface.Presenters.SoilDownloadPresenter.GetASRISSoilsAsync(IProgress`1 progress, ProgressReportModel report) in ApsimNG\Presenters\SoilDownloadPresenter.cs:line 431
```

Also, the lat and lon function works but the location name option does not.

### What Operating System are you on?

Windows

**Comment by par456** [comment]
Can you tell me what lat/long you entered for that error? Most likely that particular soil has slightly different xml from others and we need to check what's different about it.

**Comment by BrianCollinss** [comment]
@Keith-Pembleton

**Comment by BrianCollinss** [comment]
@par456 Keith used the Wheat example to download the soil. The recent updates fixed the issue with lat and long, but the location name option does not work.

**Comment by par456** [comment]
Look like google has put the API that we used for getting lat/long from placename behind a paywall now: 

```
You must enable Billing on the Google Cloud Project at https://console.cloud.google.com/project/_/billing/enable Learn more at https://developers.google.com/maps/gmp-get-started
```

We might want look for an alternative solution for this.

**Comment by zur003** [comment]
Perhaps we could consider [Nominatum](https://nominatim.org/), which is based on OpenStreetMap, for geocoding.

In the (currently stalled) SoilMapp app, I use [ESRI ArcGIS geocoding](https://developers.arcgis.com/rest/geocode/api-reference/overview-world-geocoding-service.htm), but SoilMapp has (and requires) its own ESRI licensing.

**Comment by zur003** [comment]
This same issue affects the geocoding for weather downloads,, where it fails without any warnings or messages. This needs to be fixed.

Google has changed their policy, so that the use of their geocoding services now requires that the access key be associated with a billing account. I don't know who "owns" the key that we have been using (it actually may be me, but if so I recall none of the details). Since we don't know the owner, I don't think we can associate that key with a billing account, so it's effectively become unusable.

The APSIM Initiative could obtain a new Google geocoding access key. This shouldn't normally lead to any charges, if I understand correctly, as Google provides US$200 "free" access each month. This would probably be the simplest approach if the Initiative already has a Google billing account.

An alternative would be to use [Nominatim](https://nominatim.org/). I think we could comply with their "occasional use" usage policy (although I'm not sure where we'd place the attribution). This is a good alternative if you want to stick with "open source" tools. We could even use it to set up our own geocoding service, although apparently it takes a few days to import all the data.

@hol353  - What do you see as the best alternative?

**Comment by hol353** [comment]
We have a Google A/C so going with a new geocoding key might be the simplest.

**Comment by ric394** [comment]
resolved by #9462. Closing.
